client/src/components/advancedSelectWidget/AdvancedSelect.js

```diff
@@ -275,20 +277,20 @@ const AdvancedSelect = ({
       fetchResultsDebounced(searchTerms)
     }
   }, [fetchType, fetchResults, fetchResultsDebounced, searchTerms])
-
   useEffect(() => {
     if (doReset) {
       setIsLoading(false)
       setShowOverlay(false)
       setFilterType(firstFilter)
-      setSearchTerms(selectedValueAsString)
+      if (!applySearchTerms) {
+        setSearchTerms(selectedValueAsString)
+      }
       setResults({})
```

Comment:
This seems counterintuitive: if it does not apply, set it. I would expect: "if it does apply, set it", i.e. the opposite of the current condition.